mpc-core/src/protocols/rep3/fieldshare.rs

```diff
@@ -4,16 +4,54 @@
 
 use super::id::PartyID;
 use ark_ff::PrimeField;
-use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize, Compress, Validate};
 use itertools::Itertools;
+use serde::{Deserialize, Serialize};
 
 /// This type represents a replicated shared value. Since a replicated share of a field element contains additive shares of two parties, this type contains two field elements.
-#[derive(Debug, Clone, PartialEq, Eq, Hash, CanonicalSerialize, CanonicalDeserialize)]
+#[derive(
+    Debug,
+    Clone,
+    PartialEq,
+    Eq,
+    Hash,
+    CanonicalSerialize,
+    CanonicalDeserialize,
+    Serialize,
+    Deserialize,
+)]
 pub struct Rep3PrimeFieldShare<F: PrimeField> {
+    #[serde(serialize_with = "ark_se", deserialize_with = "ark_de")]
     pub(crate) a: F,
+    #[serde(serialize_with = "ark_se", deserialize_with = "ark_de")]
     pub(crate) b: F,
 }
 
+// TODO unify with the other implementations
+
+/// Serialize an object with ark serialization, to be used with serde.
+/// `#[serde(serialize_with = "ark_se", deserialize_with = "ark_de")]`
+pub(crate) fn ark_se<S, A: CanonicalSerialize>(a: &A, s: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    let mut bytes = vec![];
+    a.serialize_with_mode(&mut bytes, Compress::Yes)
```

Comment:
Think for fields compress makes not too much sense, but I guess it should be a no-op for these?

Reply:
compress is ignored in the field element implementation:
#[inline]
    fn serialize_with_mode<W: ark_std::io::Write>(
        &self,
        writer: W,
        _compress: Compress,
    ) -> Result<(), SerializationError> {
        self.serialize_with_flags(writer, EmptyFlags)
    }